runners/portability/java/src/main/java/org/apache/beam/runners/portability/PortableRunner.java

```diff
@@ -203,11 +146,33 @@ public PipelineResult run(Pipeline pipeline) {
           prepareJobResponse.getArtifactStagingEndpoint();
       String stagingSessionToken = prepareJobResponse.getStagingSessionToken();
 
+      ImmutableList.Builder<StagedFile> filesToStageBuilder = ImmutableList.builder();
+      for (Map.Entry<String, RunnerApi.Environment> entry :
+          pipelineProto.getComponents().getEnvironmentsMap().entrySet()) {
+        for (RunnerApi.ArtifactInformation info : entry.getValue().getDependenciesList()) {
+          if (BeamUrns.getUrn(RunnerApi.StandardArtifacts.Types.FILE).equals(info.getUrn())) {
+            RunnerApi.ArtifactFilePayload filePayload;
+            try {
+              filePayload = RunnerApi.ArtifactFilePayload.parseFrom(info.getPayload());
+            } catch (InvalidProtocolBufferException e) {
+              throw new RuntimeException("Error parsing artifact file payload.", e);
+            }
+            filesToStageBuilder.add(
+                StagedFile.of(new File(filePayload.getLocalPath()), filePayload.getStagedName()));
+          } else {
+            throw new RuntimeException(
+                String.format("expect file artifact type but %s", info.getUrn()));
```

Comment:
"Unsupported artifact type ..." might be a better error here.

Reply:
done.